src/Algebra/Graph/Bipartite/AdjacencyMap.hs

```diff
@@ -709,6 +817,113 @@ biclique xs ys = let sxs = Set.fromList xs
+-- star x ys     == 'connect' ('leftVertex' x) ('vertices' [] ys)
+-- @
+star :: (Ord a, Ord b) => a -> [b] -> AdjacencyMap a b
+star x ys = overlay (leftVertex x) (edges [ (x, y) | y <- ys ])
+
+-- | The /stars/ formed by overlaying a list of 'star's. An inverse of
+-- 'leftAdjacencyList'.
+-- Complexity: /O(L log(L))/ time, where /L/ is the total size of the input.
+--
+-- @
+-- stars []                      == 'empty'
+-- stars [(x, [])]               == 'leftVertex' x
+-- stars [(x, [y])]              == 'edge' x y
+-- stars [(x, ys)]               == 'star' x ys
+-- stars                         == 'overlays' . 'map' ('uncurry' 'star')
+-- 'overlay' (stars xs) (stars ys) == stars (xs ++ ys)
+-- @
+stars :: (Ord a, Ord b) => [(a, [b])] -> AdjacencyMap a b
+stars = overlays . map (uncurry star)
+
+-- | Construct a /mesh/ graph from two lists of vertices.
+-- Complexity: /O(L1 * L2 + L1 log(L1) + L2 log(L2))/ time, where /L1/ and
+-- /L2/ are the lengths of the given lists.
+-- 
+-- @
+-- mesh xs []             == 'empty'
+-- mesh [] ys             == 'empty'
+-- mesh [x] [y]           == 'leftVertex' (x, y)
+-- mesh [1, 2] [\'a\', \'b\'] == 'biclique' [(1, \'a\'), (2, \'b\')] [(1, \'b\'), (2, \'a\')]
+-- mesh [1, 1] [\'a\', \'b\'] == 'biclique' [(1, \'a\'), (1, \'b\')] [(1, \'a\'), (1, \'b\')]
+-- @
+mesh :: (Ord a, Ord b) => [a] -> [b] -> AdjacencyMap (a, b) (a, b)
+mesh xs ys = box (path (fromList xs)) (path (fromList ys))
+
+-- | Compute the /Cartesian product/ of bipartite graphs. The vertices are
+-- sorted into parts using given combinators. Values returned by combinators
+-- are not necessarily distinct.
+-- Complexity: /O(s1 * s2 * log(s1 * s2))/ time and /O(s1 * s2)/ memory, where
+-- /s1/ and /s2/ are the sizes of the given graphs.
+--
+-- There's also a shorthand 'box' for @boxWith (,) (,) (,) (,)@.
+--
+-- @
+-- 'box' ('path' [0,1]) ('path' [\'a\',\'b\']) == 'edges' [ ((0,\'a\'), (0,\'b\'))
+--                                                    , ((0,\'a\'), (1,\'a\'))
+--                                                    , ((0,\'b\'), (1,\'b\'))
+--                                                    , ((1,\'a\'), (1,\'b\')) ]
+-- @
+-- Up to an isomorphism between the resulting vertex types, this operation
+-- is /commutative/, /associative/, /distributes/ over 'overlay', has singleton
+-- graphs as /identities/ and 'empty' as the /annihilating zero/. Below @~~@
+-- stands for the equality up to an isomorphism, e.g. @(x, ()) ~~ x@
+--
+-- @
+-- box x y               ~~ box y x
+-- box x (box y z)       ~~ box (box x y) z
+-- box x ('overlay' y z)   == 'overlay' (box x y) (box z)
+-- box x ('vertex' ())     ~~ x
+-- box x 'empty'           ~~ 'empty'
+-- 'vertexCount' (box x y) <= 'vertexCount' x * 'vertexCount' y
+-- 'edgeCount'   (box x y) <= 'vertexCount' x * 'edgeCount' y + 'edgeCount' x * 'vertexCount' y
+-- @
+boxWith :: forall a b c d e f. (Ord a, Ord b, Ord c, Ord d, Ord e, Ord f) =>
```

Comment:
Can we get rid of `forall a b c d e f` in this type signature? It's best to avoid such top-level `forall`s since they sneak into Haddock. In this case, I think it can just be dropped.